thoth/prescriptions_refresh/handlers/image_analysis.py

```diff
@@ -0,0 +1,206 @@
+      runtime_environments:
+        operating_systems:
+        - name: {os_name}
+          version: {os_version}
+        python_version: =={python_version}
+        base_images:
+          not:
+          - {image}
+    match:
+      state:
+        resolved_dependencies:
+        {resolved_dependencies}
+    run:
+      stack_info:
+      - type: INFO
+        message: >-
+          Found predictive stack image that can be used with these dependencies
+        link: {link}
+      advised_manifest_changes:
+      - file: .thoth.yaml
+        patch:
+        - name: base_image
+          value: {link}
+"""
+
+USER_API_HOST = os.environ["THOTH_USER_API_HOST"]
+
+
+def _get_latest_image_analyzed_info(image: str) -> Optional[Dict[str, Any]]:
+    """Get latest image analyzed information."""
+    url = f"http://{USER_API_HOST}/api/v1/container-images"
+    response = requests.get(url, params={"image_name": f"{QUAY_URL}/thoth-station/{image}"})
+
+    if response.status_code == 200:
+        results = response.json()
+        return results[0]
+    else:
+        return None
+
+
+def _get_requirement_files_from_image_analysis(
+    package_extract_document_id: str,
+) -> Tuple[Optional[Dict[str, Any]], Optional[Dict[str, Any]]]:
+    """Get requiremens files from image analysis result."""
+    url = f"http://{USER_API_HOST}/api/v1/analyze"
+    response = requests.get(url, params={"analysis_id": package_extract_document_id})
+
+    if response.status_code == 200:
+        document = response.json()
+
+        # Get result and requirements files
+        result = document["result"]
+        pipfile_dict = result["aicoe-ci"].get("requirements")
+        pipfile_lock_dict = result["aicoe-ci"].get("requirements_lock")
+
+        return pipfile_dict, pipfile_lock_dict
+
+    elif response.status_code == 404:
+        _LOGGER.warning(f"Document {package_extract_document_id} does not exists.")
+        return None, None
+
+    else:
+        return None, None
+
+
+def _create_resolved_dependencies_section(pipfile: Dict[str, Any], pipfile_lock: Dict[str, Any]) -> str:
+    """Create resolved dependencies section for adviser unit."""
+    resolved_dependencies = ""
+
+    pipfile = Pipfile.from_dict(pipfile)
+    pipfile_lock = PipfileLock.from_dict(pipfile_lock)
+
+    required_packages = []
+
+    for package in pipfile.packages.packages:
+        version = pipfile_lock.packages[package].version
+        required_packages.append({"name": package, "version": version})  # Includes ==
+
+    line = 0
+    for package in required_packages:
+        if line == 0:
+            resolved_dependencies += f"- name: {package['name']}\n"
+        else:
+            resolved_dependencies += f"        - name: {package['name']}\n"
+
+        resolved_dependencies += f"          version: {package['version']}\n"
+        line += 1
+
+    return resolved_dependencies
+
+
+def _generate_prescription_name(image, tag) -> str:
+    """Generate custom prescription name.
```

Comment:
We recently added a general method for generating prescription names here: https://github.com/thoth-station/prescriptions-refresh-job/blob/dd2e53e87941908e54031ce776ed7725c2168b64/thoth/prescriptions_refresh/prescriptions.py#L202

Reply:
Nice, I didn't see it sorry :D

Reply:
No problem :slightly_smiling_face: